src/mapping/nav2_map_server/src/map_reps/map_loader.cpp

```diff
@@ -16,29 +16,34 @@
 
 #include "nav2_map_server/map_loader.hpp"
 
+namespace nav2_map_server
+{
+
 // Factory Class for loading new map types
-BaseMapLoader * MapLoader::createMap(
-  std::string mapType,
-  rclcpp::Node::SharedPtr n, std::string filename)
+BaseMapLoader * MapLoader::CreateMap(
+  std::string map_type,
+  rclcpp::Node::SharedPtr node, std::string file_name)
 {
-  if (mapType == "occupancy") {
-    return new OccGridLoader(n, filename);
-  } else if (mapType == "gridmap") {
-    return new OccGridLoader(n, filename);  // TODO(bpwilcox): Substitute with GridMapLoader
+  if (map_type == "occupancy") {
+    return new OccGridLoader(node, file_name);
+  } else if (map_type == "gridmap") {
+    return new OccGridLoader(node, file_name);  // TODO(bpwilcox): Substitute with GridMapLoader
   } else {
-    fprintf(stderr, "[ERROR] [map_server]: Cannot Load Map of Type '%s'\n", mapType.c_str());
+    fprintf(stderr, "[ERROR] [map_server]: Cannot Load Map of Type '%s'\n", map_type.c_str());
     exit(-1);
   }
 }
 
-BaseMapLoader * MapLoader::createMap(std::string mapType)
+BaseMapLoader * MapLoader::CreateMap(std::string map_type, rclcpp::Node::SharedPtr node)
```

Comment:
The constructor for the OccGridLoader calls the load function which would expect a valid filename otherwise throw an error, so I would have to add a check for an empty string. My reasoning for the overloaded function was to allow the object to be initialized without immediately loading a file.